source-azure-blob-storage/go.mod

```diff
@@ -0,0 +1,49 @@
+module github.com/estuary/source-azure-blob-storage
```

Comment:
Our convention has been to have all go connectors use the same top-level `go.mod` file: https://github.com/estuary/connectors/blob/main/go.mod

I'm not necessarily saying that is better or worse than each connector defining its dependencies separately, but its what we've been doing and I don't want to diverge from that right now. So you should be able to add the necessary dependencies to the main `go.mod`.

For example, see [this commit](https://github.com/estuary/connectors/commit/28ee7a8cfcafb123d059359625f8e574fda84973) where we added a filesource connector and updated `go.mod` with what it needed.